mediatek: mt7623: fix thermal zone

anonimou0 opened the pull request:
As discussed on #9396 , using @VA1DER patch posted there and the tests done, I'm trying to create this pull request.

Text from @VA1DER:

The CPU will be throttled all the way down to 98MHz if the temperature rises even a degree above the trip point, and I have further discovered that if the internal temperature of the device is above the first trip point temperature when it boots, 

The patch deletes the first two trip points and cooling maps. The throttling temperature will then be at 87°C, which is still a low enough temperature for ARM devices to not be in the real danger zone, and gives some operational headroom.

dangowrt commented:
Both active and passive trip points are desirable to have. Simply dropping them is not an option. Instead, we should swap the order (ie. first active == fan, then passive == throttling) and maybe raise the trip-point temperatures a little if needed.
On the formal side this PR also lacks a meaningful commit message as well as `Signed-off-by:` line.

anonimou0 commented:
Followed your comment, @dangowrt , and swapped both active and passive trip points

Also raised 10oC on the active trip point and 30oC on the passive trip point.

Tested on UniElec U7623-02 eMMC (512M RAM)

dangowrt reviewed (changes requested):
Please squash the 3 commits into a single one (eg. using `git rebase -i origin/master` and then using the `fixup` or `squash` commands in the editor) and also write a meaningful commit message including Signed-off-by line (also in the commit for openwrt.git, not just the in the patch file you are adding).
The commit title should start with `mediatek: mt7623: ...`.

See https://openwrt.org/submitting-patches

anonimou0 commented:
Sorry for bothering and thanks for the guidance, @dangowrt .

I believe I successfully made the changes requested.

robimarko commented:
Unless there is an active cooling device like a fan then active and passive are both gonna just end up throttling and that's it.

robimarko commented:
Thats a different thing then

VA1DER commented:
> Both, the U7623 and the BPi-R2 do have a connector for a PWM-driven fan

Where does the U7623 even have a fan connector?   And on the R2, has anyone ever got the fan PWM to work?

In any case, there is certainly no way of activating a fan that is _common_ to both the U7623 and the BPI-R2.  I don't believe the appropriate place to have .dts entries for U7623/BPI-R2 fans is in their common platform .dtsi.  If it were appropriate to have all three trip points in the .dtsi, it still wouldn't be appropriate to have them all tied to identical cooling maps.

I still suggest the appropriate course of action is to remove two of the trip points and their associated (identical) cooling maps from the .dtsi, and if we get the PWM's working with fans, incorporate what we need into the individual device .dts's

dangowrt commented:
@graysky2 Ok, agreed, let's just remove the lowest trip point for now and have new first trippoint at 77 deg C called `passive` with CPU as cooling device, because that makes most sense.

anonimou0 commented:
> Ok, agreed, let's just remove the lowest trip point for now and have new first trippoint at 77 deg C called passive with CPU as cooling device, because that makes most sense.

Unless I got something wrong, the current push should provide that.

frank-w commented:
- It seems you're touching 2 files.
- Please squash commits into 1 as this is a single change
- signed off tag have to use real name

anonimou0 commented:
I will, eventually, get this right, with the rebases.... Sorry for the noise.

Does this look good now?

anonimou0 commented:
Hello.

Just checking if any more changes are required.

Thanks

anonimou0 commented:
```
Author name (anonimou) need to be your real name 'firstname lastname'
Commit subject line seems ok (mediatek: mt7623: thermal zone fix The fix was proposed by @VA1DER at issue 9396.)
Signed-off-by is missing or doesn't match author (should be 'Signed-off-by: anonimou <anonimou_eu@hotmail.com>')
```

Should this be the right one? Should I change the github name to match what I signed off?

anonimou0 commented:
Pushed with the right .gitconfig, now?

Sorry for causing all this issue. squashing the commits are being harder than I could figure, hence the force push.

frank-w commented:
Imho commit title is too long, usally should take max 75 chars like all lines.

You could change it like the title of this issue

"mediatek: mt7623: Thermal zone fix"

And put your current title down in the description with linebreaks after max 75 chars.

anonimou0 commented:
Sorry for the long delay on this.

Following @frank-w suggestion, changed the commit messages.

anonimou0 commented:
Hello guys. Not pushing at all.. You are probably busy with 22.03 release.

But just want to check if any other change is required.

dangowrt commented:
The commit message has too long lines (please add line breaks) and the commit title should be 'mt7623: fix thermal zone' or something like that. There shouldn't be a capital letter at the beginning and the commit title should be in imperative form.
Please also do not remove the `active` trip point, it can still be useful on boards which do have a PWM controller fan.

anonimou0 commented:
Which cooling maps should stay, then? `map0`and `map1`?

Did a new push with those.

anonimou0 commented:
Hello. 

Just a quick check if I'm still missing something.

Thanks

dangowrt commented:
> Which cooling maps should stay, then?

Both. Just because your board doesn't have a fan (`active` trip point) it may still be relevant to others. Bananapi R2 has a PWM output for a fan which could (and should) be wired up as a cooling device on the lowest trip-point, so please do not remove it.

Just *swapping* `active` and `passive` types should do the trick for boards with and without a fan.

anonimou0 commented:
How about these comments

https://github.com/openwrt/openwrt/pull/9778#issuecomment-1114368166

and 

https://github.com/openwrt/openwrt/pull/9778#issuecomment-1111186735

Should I just increase the trip point temperatures, keeping everything else as is?

Ansuel commented:
@anonimou0 IMHO we should just increase the passive trip point.... seems a much safer approach to this without too much delta.... Also pls refresh or produce these patch from a linux clone.

anonimou0 commented:
Thank you @Ansuel .

When you say to `produce these patch from a linux clone.` you mean patching the dtsi file itself? In here or on the linux kernel itself? Could you clarify? Thanks.

Ansuel commented:
@anonimou0 ideally you should make the patch on a kernel linux clone... but in theory if the patch apply quilt should fix it so a 
make target/linux/refresh should produce correct patch anyway

anonimou0 commented:
I think I got this right. 

Cloned linux repo, and got the latests file. Changed and created the patch.

Also, sorry for closing and reopening the pull request. I also updated my openwrt fork and it closed automatically.

Ansuel commented:
@anonimou0 how you created the patch from the linux clone ?

anonimou0 commented:
```
 2002  git clone https://github.com/torvalds/linux
 2003  cd linux/arch/arm/boot/dts/
 2004  ls
 2005  vim mt7623.dtsi 
 2006  git diff mt7623.dtsi
 2007  git diff mt7623.dtsi > ~/193-dts-mt7623-thermal_zone_fix.patch
```

On the generated file, there was a index: `index 25d31e40a553..14b6a6442702 100644` which I wasn't sure if I should keep or not. Removed.

Then added the comment headers on the file, committed and pushed.

Ansuel commented:
(if the file was not present commit if before this change to any commit)
then make the change and commit it
and generate the patch with git format-patch HEAD^

anonimou0 commented:
Can you check now, @Ansuel ?

Should all those headers be kept?

Ansuel commented:
@anonimou0 yes if in doubt just post the full patch on pastebin and post the link here... anyway you should not modify the output from git format-patch

anonimou0 commented:
I pushed with the output of got format-patch already

Ansuel commented:
@anonimou0 REALLY NICE! now just /target/linux/refresh push the final version and we are good to go !

anonimou0 commented:
@Ansuel there is no  /target/linux/refresh file and make target/linux/refresh and make target/linux/update both fail without a recipe

Ansuel commented:
what do you mean without a recipe?

anonimou0 commented:
The command was `make target/linux refresh push`. Running it now.

anonimou0 commented:
> what do you mean without a recipe?

My bad. I was running the wrong command.

Should I finally push just the patch file or all of them?

```
	modified:   target/linux/generic/backport-5.15/005-v5.17-01-Kbuild-use-Wdeclaration-after-statement.patch
	modified:   target/linux/generic/backport-5.15/005-v5.17-02-Kbuild-move-to-std-gnu11.patch
	modified:   target/linux/generic/backport-5.15/005-v5.17-03-Kbuild-use-std-gnu11-for-KBUILD_USERCFLAGS.patch
	modified:   target/linux/generic/backport-5.15/011-kbuild-export-SUBARCH.patch
	modified:   target/linux/generic/hack-5.15/204-module_strip.patch
	modified:   target/linux/mediatek/patches-5.15/193-dts-mt7623-thermal_zone_fix.patch
```

Ansuel commented:
@anonimou0 only the relevant patch pls

anonimou0 commented:
Done? Maybe?

Ansuel commented on target/linux/mediatek/patches-5.15/193-dts-mt7623-thermal_zone_fix.patch:
```diff
@@ -0,0 +1,49 @@
+zone, and gives some operational headroom.
+
+Signed-off-by: Bruno Umuarama <anonimou_eu@hotmail.com>
+￼
```
The only thing I can't understand is this... O.o ?

anonimou0 replied:
Shouldn't that be included?

https://github.com/openwrt/openwrt/pull/9778#issuecomment-1110373581

Ansuel replied:
No, not the sob tag... the char at the next line... can you remove it and just remove that line? 

`￼` <-- this thing

anonimou0 replied:
Weird. It is not showing here. Possibly form many copy and pastes. Removed and committed.

anonimou0 committed:
mediatek: mt7623: fix thermal zone

Raising the temperatures for passive and active trips. @VA1DER
proposed at issue 9396 to remove passive trip. This commit relates to
his suggestion.

Without this patch. the CPU will be throttled all the way down to 98MHz
if the temperature rises even a degree above the trip point, and it was
further discovered that if the internal temperature of the device is
above the first trip point temperature when it boots then it will start
in a throttled state and even
$ echo disabled > /sys/class/thermal/thermal_zone0/mode
will have no effect.

The patch increases the passive trip point and active cooling map. The
throttling temperature will then be at 77°C and 82°C, which is still a
low enough temperature for ARM devices to not be in the real danger
zone, and gives some operational headroom.

Signed-off-by: Bruno Umuarama <anonimou_eu@hotmail.com>

Ansuel commented:
@anonimou0 it was a long journey but we manage to make it. Thanks for following instructions and not getting annoyed by me ahahah.

anonimou0 commented:
@Ansuel, no annoyance at all. Actually, I felt the other way, annoying you guys. Thanks for the patience. Learned a lot on the way.

Thanks @Ansuel , @dangowrt , @frank-w , @VA1DER
